tests/Integration/Behaviour/Features/Context/Domain/Product/UpdateAttachmentFeatureContext.php

```diff
@@ -103,7 +105,14 @@ public function setAssociatedProductAttachments(string $productReference, string
             $attachmentIds[] = $this->getSharedStorage()->get($attachmentReference);
         }
 
-        $this->setProductAttachments($this->getSharedStorage()->get($productReference), $attachmentIds);
+        try {
+            $this->getCommandBus()->handle(new SetAssociatedProductAttachmentsCommand(
+                $this->getSharedStorage()->get($productReference),
+                $attachmentIds
+            ));
+        } catch (DomainException $e) {
```

Comment:
The exception is too wide here, either use `ProductException` to limit it to the Product domain, allowing unexpected other exceptions to be thrown Because we usually only test lastException when we expect it, but never when nothing should raise, so quite often these caught exceptions just go unnoticed (until something worse bugs down the road 😅 )

But I recommended like in another PR, if there is no test related to this caught exception, we might as well not catch anything

Reply:
This makes me think, maybe we should add a safeguard on lastException If the next behat check doesn't handle to check it, it probably means that we didn't expect this and we could throw a RuntimeException like `Some exception has been caught that did not seem to be expected since no assertions is related to it` What do you think?

Reply:
It happens already. If some exception is thrown and is not handled by next scenario it will fail and tell something like "probably it was caused by {blabla}Exception"

Reply:
Anyway I removed try-catch here as no exceptions are asserted in scenarios

Reply:
We have this safeguard indeed but only AfterScenario, the idea would to raise an error much sooner, like in the very next step To avoid having 4-5 steps that run in an unexpected state and start bugging, and sometimes you don't realize right away that the error happened a few steps earlier

It would also force us to correctly set the expected exception everywhere they are wanted And as you say we should improve the display as the gigantic message is not really easy to understand, I even believe sometimes it kind of "loops" into the stacktrace and display the same thing several times

Anyway it's a parallel improvement 🙂